### ISSUE TITLE
prometheus node_exporter should be on all nodes

### ISSUE THREAD

**Issue by jayunit100** [helm/charts#1602]
**Is this a request for help?**:

no

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
 
bug 

**Version of Helm and Kubernetes**:

ubuntu@ip-10-0-22-242:~$ /opt/helm-exec version
Client: &version.Version{SemVer:"v2.5.0", GitCommit:"012cb0ac1a1b2f888144ef5a67b8dab6c2d45be6", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.5.0", GitCommit:"012cb0ac1a1b2f888144ef5a67b8dab6c2d45be6", GitTreeState:"clean"}
ubuntu@ip-10-0-22-242:~$ kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.6", GitCommit:"7fa1c1756d8bc963f1a389f4a6937dc71f08ada2", GitTreeState:"clean", BuildDate:"2017-06-16T18:34:20Z", GoVersion:"go1.7.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.6", GitCommit:"7fa1c1756d8bc963f1a389f4a6937dc71f08ada2", GitTreeState:"clean", BuildDate:"2017-06-16T18:21:54Z", GoVersion:"go1.7.6", Compiler:"gc", Platform:"linux/amd64"}

**Which chart**:

prometheus

**What happened**:

Only one node_exporter is deployed, the one that is on the prometheus master, with the alerter and master : 

```
ubuntu@ip-10-0-22-242:~$ kubectl describe pod waxen-crocodile-promethe-2633427841-f0cwt | grep Image
    Image:              prom/prometheus:v1.1.0
    Image ID:           docker-pullable://prom/prometheus@sha256:e37f0f84f913ef7abcd8c90ae8fc22f784f67fdce58d6f42d7c3c4ec57100098
    Image:              prom/alertmanager:v0.4.2
    Image ID:           docker-pullable://prom/alertmanager@sha256:f077486dfbc1c79a49373741f7bce744a455f42906c29560b3f40e842156a802
    Image:              prom/node-exporter
    Image ID:           docker-pullable://prom/node-exporter@sha256:ffc3e0adf58771cb2097aa0b074a6fe68b4925ba75c2e1c41f41ae656eebee11
```
**What you expected to happen**:

Node exporters should be fired up on all nodes, not just one.  Or at least, the scraper should attempt to pull metrics from 9091 on all nodes by default.

**How to reproduce it** 

(as minimally and precisely as possible):

This is standard behaviour for the chart - requires no special reproducer - im just suggesting that its a bit of a usability bug since exporting metrics from one node creates an asymmetric report .  We should either 
- export no node metrics and let the user completely wire the node exporters, or else, 
- we should export node metrics on all nodes, so the user has to do nothing. 
- Or force the user to choose somehow.

**Anything else we need to know**:

**Comment by jayunit100** [comment]
cc @jimmidyson or any other prometheus folks:  
- how should we manage node exporters ?  Right now we seem to run one.  Why exactly one?  Seems like N=(# of nodes) or N=0 is more consistent.
- I realize there are some suggestions out there that you don't run node_exporters in containers (https://github.com/prometheus/node_exporter/pull/376) .  .

**Comment by naveensrinivasan** [comment]
I agree the nodeexporter running within docker container does not scrape new volumes. It is not advisable to run within a container.

**Comment by JeanMertz** [comment]
I'm a bit confused. I've just set up our Prometheus set-up on our cluster, but AFAIK the node exporter is deployed as a daemon set, and it is correctly started on _all_ nodes in our cluster. We've got 5 nodes right now, and 5 pods running, one on each node, managed by the node exporter DS.

Was this recently fixed, or am I misunderstanding the reported issue?

This is separate from the question if we want to run node exporters in containers or not, but a word of warning here: we are running on GKE, and rely heavily on preemptible machines, which are recycled at least once every 24 hours, all of this is running on top of Googles' "container OS". Running this utility on the nodes itself would probably be possible, but certainly more difficult.
